app/controllers/candidate_interface/degrees/degree/review_controller.rb

```diff
@@ -34,6 +35,20 @@ def application_form_params
         def redirect_to_old_degree_flow_unless_feature_flag_is_active
           redirect_to candidate_interface_new_degree_path unless FeatureFlag.active?(:new_degree_flow)
         end
+
+        def set_completed_if_only_foundation_degrees
+          if only_foundation_degrees?
+            current_application.update!(degrees_completed: nil)
+          end
+        end
+
+        def only_foundation_degrees?
+          degree_type = []
+          current_application.application_qualifications.degrees.pluck(:qualification_type).map do |degree|
+            degree_type << Hesa::DegreeType&.find_by_name(degree)&.level
```

Comment:
As you are using map I don't think you need to build the `degree_type` array manually. e.g.

```
degree_types = current_application.application_qualifications.degrees.pluck(:qualification_type).map do |degree|
  Hesa::DegreeType&.find_by_name(degree)&.level
end
```

Reply:
Thanks Steve, I've fixed !